channels/urbdrc/client/data_transfer.c

```diff
@@ -2319,12 +2319,9 @@ void* urbdrc_process_udev_data_transfer(void* arg)
 	pdev = udevman->get_udevice_by_UsbDevice(udevman, UsbDevice);
 	if (pdev == NULL || pdev->isSigToEnd(pdev))
 	{
-		if (transfer_data)
-		{
-			if (transfer_data->pBuffer)
-				zfree(transfer_data->pBuffer);
-			zfree(transfer_data);
-		}
+		if (transfer_data->pBuffer)
```

Comment:
that looks more like an issue with cppcheck.
There is no way to know the function will always be called with a non NULL argument (although it currently is the case)